File: java/oop/AcccountWithCustomException/src/com/techlabs/account/Account.java
package com.techlabs.account;

public class Account {
	private int acnt_no;
	private String name;
	private double balance=500;
	private static int count=1000;

	
	static{
		count=1000;
		System.out.println("This is Static Block...");
	}
	public Account(int acnt_no,String name,double balance){
		this(acnt_no,name);
		this.balance = balance;
	}
	public Account(int acnt_no,String name){
		this();
		this.acnt_no = acnt_no;
		this.name = name;	
	}
	public Account(){
		count+=1;
		System.out.println("object created...................");
		System.out.println("This is default constructor......");
	}
	
	public String getName(){
		return this.name;
	}
	
	public int getAcntNo(){
		return this.acnt_no;
	}
	public int getCount(){
		return Account.count;
	}
	
	public static int HeadCount(){
		return Account.count;
	}
	public void deposit(double amt){
		this.balance+=amt;
		System.out.println("Ammout successfully depostied");
	}
	
	public void withdraw(double amt) throws InsuficientFundException {
		
	
		if((this.balance-amt)<=500){
		
			throw new InsuficientFundException("insufficient fund..",this);
			
		}else{
		
			this.balance=this.balance-amt;
		}
	}
	
	@Override
	public String toString(){
		//String s1=super.toString();
		//System.out.println(s1);
		return String.format("Name:"+name+ "\nAccount-No:"+acnt_no+"\nBalance:"+balance);
	}
	

}

File: dotNet/csharp/AccountAppWithCustomException/AccountAppWithCustomException/Account.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AccountAppWithCustomException
{
    class Account
    {
        private int acnt_no;
        private string name;
        private double balance = 500;
        private static int count;

        static Account()
        {
            count = 1000;
            Console.WriteLine(" Static Block called once....");
        }

        public Account(int acnt_no, string name, double balance) : this(acnt_no, name)
        {
            this.balance = balance;
        }
        public Account(int acnt_no, string name) : this()
        {
            this.acnt_no = acnt_no;
            this.name = name;
        }
        public Account()
        {
            count += 1;
            Console.WriteLine("Default constructor  called");
        }

        public string Name
        {
            get
            {
                return this.name;
            }

        }

        public int AccntNo
        {
            get
            {
                return this.acnt_no;
            }

        }

        public int Count
        {
            get
            {
                return Account.count;
            }

        }
        public static int HeadCount
        {
            get
            {
                return Account.count;
            }

        }

        public void Deposite(double amt)
        {
            this.balance += amt;
            Console.WriteLine("Amount deposited successfully..");
        }
        public void Withdraw(double amt)
        {
            if ((this.balance - amt) <= 500)
            {
                throw new InsufficientFundExceptin("Minimum balance limit exceeded...",this);
            }
            else
            {
                Console.WriteLine("Withdraw done..");
                this.balance -= amt;
            }
        }
        public override string ToString()
        {
            return String.Format("Name:" + name + "\nAccount-No:" + acnt_no + "\nBalance:" + balance + "\nCount :" + this.Count + "\nHeadCount :" + HeadCount);
        }
    }
}
